configure.ac

```diff
@@ -771,7 +771,7 @@ dnl Check for libminiupnpc (optional)
 if test x$use_upnp != xno; then
   AC_CHECK_HEADERS(
     [miniupnpc/miniwget.h miniupnpc/miniupnpc.h miniupnpc/upnpcommands.h miniupnpc/upnperrors.h],
-    [AC_CHECK_LIB([miniupnpc], [main],[MINIUPNPC_LIBS=-lminiupnpc], [have_miniupnpc=no])],
+    [AC_CHECK_LIB([miniupnpc], [upnpDiscover], [MINIUPNPC_LIBS=-lminiupnpc], [have_miniupnpc=no])],
```

Comment:
I don't see how this is related to Clang 7, this is checking for the visibility of a symbol. This might be needed on your version of upnp but if so that should be another PR in my opinion.

Reply:
The commit message of that change reads:

```
    Do not check for main() in libminiupnpc
    
    main() { main(); } causes "infinite recursion" compilation warning
    which with -Werror fails the check.
```

Maybe that is not clear enough, let me elaborate: `AC_CHECK_LIB([lib], [func], ...)` generates a test program like `main() { func(); ... }`  and tries to link it with `-llib`. The idea being that if `func()` is not present in `lib` it will fail to link with an undefined reference error. So the current usage leads to a test program like `main() { main(); ... }` which generates a legit compiler warning of "infinite recursion". At least Clang 7 emits that warning, maybe other compilers do so too.

Anyway, I think checking for function `main()` in `libminiupnpc` is a misuse of `AC_CHECK_LIB` because `main()` does not exist in `libminiupnpc` and it passes because it relies on how the check is performed internally. If `AC_CHECK_LIB` starts using another method of checking if a function is present in a library (e.g. `nm(1)`) then the check may as well fail.

Reply:
That makes sense, thanks for clarifying.